File: flutter_app/lib/screens/notification_warning_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../theme/app_colors.dart';
import '../widgets/care_header.dart';
import '../widgets/care_card.dart';

class NotificationWarningScreen extends StatelessWidget {
  const NotificationWarningScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.pageBg,
      body: Column(
        children: [
          CareHeader(
            title: 'Notifications',
            onBack: () => context.pop(),
          ),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(24),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  CareCard(
                    backgroundColor: AppColors.warningBg,
                    borderColor: AppColors.warningLight,
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            Container(
                              padding: const EdgeInsets.all(10),
                              decoration: BoxDecoration(
                                color: AppColors.warningLight,
                                borderRadius: BorderRadius.circular(14),
                              ),
                              child: const Icon(
                                Icons.notifications_off,
                                color: AppColors.warningDark,
                                size: 24,
                              ),
                            ),
                            const SizedBox(width: 12),
                            const Expanded(
                              child: Text(
                                'Notifications are turned off',
                                style: TextStyle(
                                  fontSize: 20,
                                  fontWeight: FontWeight.w700,
                                  color: AppColors.warningDark,
                                ),
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 12),
                        const Text(
                          'You may miss important medication reminders and appointment alerts. '
                          'We recommend enabling notifications to stay on track.',
                          style: TextStyle(
                            fontSize: 16,
                            color: AppColors.heading,
                            height: 1.5,
                          ),
                        ),
                      ],
                    ),
                  ),
                  const SizedBox(height: 24),
                  const Text(
                    'What you might miss:',
                    style: TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.w700,
                      color: AppColors.heading,
                    ),
                  ),
                  const SizedBox(height: 16),
                  _warningItem(
                    Icons.medication,
                    'Medication Reminders',
                    'Timely alerts when your medicine is due.',
                    AppColors.primaryAction,
                  ),
                  const SizedBox(height: 12),
                  _warningItem(
                    Icons.calendar_today,
                    'Appointment Alerts',
                    'Reminders before your scheduled visits.',
                    AppColors.caregiver,
                  ),
                  const SizedBox(height: 12),
                  _warningItem(
                    Icons.person,
                    'Caregiver Messages',
                    'Updates and check-ins from your helpers.',
                    AppColors.success,
                  ),
                  const SizedBox(height: 24),
                ],
              ),
            ),
          ),
          _buildFooter(context),
        ],
      ),
    );
  }

  Widget _warningItem(IconData icon, String title, String desc, Color color) {
    return CareCard(
      child: Row(
        children: [
          Container(
            padding: const EdgeInsets.all(10),
            decoration: BoxDecoration(
              color: color.withValues(alpha: 0.1),
              borderRadius: BorderRadius.circular(14),
            ),
            child: Icon(icon, color: color, size: 24),
          ),
          const SizedBox(width: 16),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  title,
                  style: const TextStyle(
                    fontSize: 18,
                    fontWeight: FontWeight.w700,
                    color: AppColors.heading,
                  ),
                ),
                Text(
                  desc,
                  style: const TextStyle(fontSize: 16, color: AppColors.mutedText),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildFooter(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: const BoxDecoration(
        color: AppColors.white,
        border: Border(top: BorderSide(color: AppColors.border, width: 4)),
      ),
      child: SafeArea(
        top: false,
        child: Column(
          children: [
            SizedBox(
              width: double.infinity,
              height: 60,
              child: ElevatedButton(
                onPressed: () {
                  ScaffoldMessenger.of(context).showSnackBar(
                    const SnackBar(
                      content: Text('Notifications enabled'),
                      backgroundColor: AppColors.success,
                    ),
                  );
                  context.pop();
                },
                style: ElevatedButton.styleFrom(
                  backgroundColor: AppColors.primaryAction,
                  foregroundColor: AppColors.white,
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(20),
                  ),
                ),
                child: const Text(
                  'Enable Notifications',
                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.w700),
                ),
              ),
            ),
            const SizedBox(height: 12),
            SizedBox(
              width: double.infinity,
              height: 56,
              child: OutlinedButton(
                onPressed: () => context.pop(),
                style: OutlinedButton.styleFrom(
                  foregroundColor: AppColors.heading,
                  side: const BorderSide(color: AppColors.border, width: 3),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(20),
                  ),
                ),
                child: const Text(
                  'Keep Turned Off',
                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.w700),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: react-native-app/src/screens/NotificationWarningScreen.js
import React from 'react';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet, Alert } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import CareHeader from '../components/CareHeader';
import Colors from '../theme/colors';

export default function NotificationWarningScreen({ navigation }) {
  return (
    <SafeAreaView style={styles.safe}>
      <CareHeader
        title="Notifications"
        onBack={() => navigation.goBack()}
        onEmergency={() => navigation.navigate('Emergency')}
      />
      <ScrollView contentContainerStyle={styles.scroll}>
        <View style={styles.header}>
          <View style={styles.iconCircle}>
            <Ionicons name="notifications-off" size={48} color={Colors.warningDark} />
          </View>
          <Text style={styles.heading}>Notifications Are Off</Text>
          <Text style={styles.body}>
            Without notifications, you may miss important medication reminders and health alerts.
          </Text>
        </View>

        <View style={styles.warningBox}>
          <Ionicons name="alert-circle" size={24} color={Colors.warningDark} />
          <Text style={styles.warningText}>
            Turning on notifications helps ensure you never miss a medication dose or appointment.
          </Text>
        </View>

        <View style={{ height: 32 }} />
        <TouchableOpacity
          style={styles.primaryButton}
          onPress={() => {
            Alert.alert('Notifications', 'Notifications enabled (simulated)');
            navigation.goBack();
          }}
        >
          <Text style={styles.primaryText}>Enable Notifications</Text>
        </TouchableOpacity>
        <View style={{ height: 12 }} />
        <TouchableOpacity
          style={styles.secondaryButton}
          onPress={() => navigation.goBack()}
        >
          <Text style={styles.secondaryText}>Keep Turned Off</Text>
        </TouchableOpacity>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safe: { flex: 1, backgroundColor: Colors.pageBg },
  scroll: { padding: 24, flexGrow: 1 },
  header: { alignItems: 'center', marginTop: 16, marginBottom: 24 },
  iconCircle: {
    width: 96, height: 96, borderRadius: 48, backgroundColor: Colors.warningBg,
    justifyContent: 'center', alignItems: 'center', marginBottom: 16,
  },
  heading: { fontSize: 28, fontWeight: '900', color: Colors.heading, textAlign: 'center' },
  body: { fontSize: 18, color: Colors.mutedText, textAlign: 'center', lineHeight: 28, marginTop: 8 },
  warningBox: {
    flexDirection: 'row', gap: 12, padding: 20,
    backgroundColor: Colors.warningBg, borderRadius: 20,
    borderWidth: 2, borderColor: Colors.warningLight,
  },
  warningText: { fontSize: 16, color: Colors.warningDark, lineHeight: 24, flex: 1 },
  primaryButton: {
    backgroundColor: Colors.primaryAction, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
  },
  primaryText: { fontSize: 20, fontWeight: '700', color: Colors.white },
  secondaryButton: {
    backgroundColor: Colors.white, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
    borderWidth: 3, borderColor: Colors.border,
  },
  secondaryText: { fontSize: 20, fontWeight: '700', color: Colors.mutedText },
});
